app/src/main/java/nu/yona/app/api/model/AppMetaInfo.java

```diff
@@ -0,0 +1,46 @@
+package nu.yona.app.api.model;
+
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.widget.TextView;
+
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+
+import nu.yona.app.BuildConfig;
+import nu.yona.app.R;
+import nu.yona.app.YonaApplication;
+import nu.yona.app.ui.settings.SettingsFragment;
+import nu.yona.app.utils.AppConstant;
+import nu.yona.app.utils.AppUtils;
+
+import static nu.yona.app.ui.YonaActivity.getActivity;
+
+public class AppMetaInfo {
+
+    @SerializedName("operatingSystem")
+    @Expose
+    private String operatingSystem = YonaApplication.getAppContext().getString(R.string.operating_system);
+
+    @SerializedName("appVersion")
+    @Expose
+    private String appVersion  =  BuildConfig.VERSION_NAME;
+
+    @SerializedName("appVersionCode")
+    @Expose
+    private Integer appVersionCode = BuildConfig.VERSION_CODE;
+
+
+    public Integer getAppVersionCode() {
+        return appVersionCode;
+    }
+
+    public String getAppVersion() {
+        return appVersion;
+    }
+
+    public String getOperatingSystem() {
+        return operatingSystem;
+    }
+
```

Comment:
I suggest a slightly different approach. The files ``SettingsFragment.java`` and ``AppNetworkImpl.java`` contains this statement:

```
AppMetaInfo appMetaInfo = new AppMetaInfo();
```

That looks odd, as nothing goes into the constructor.

To prevent that, please do the following:
* Make a private constructor that takes three parameters: ``private AppMetaInfo (String operatingSystem, String appVersion, int appVersionCode)``
* Assign the parameters to the member variables
* Create a private static member: ``private static final AppMetaInfo theInstance = new AppMetaInfo ("ANDROID", BuildConfig.VERSION_NAME, BuildConfig.VERSION_CODE);``
* Create a public static method: ``private static AppMetaInfo()`` that returns the instance
* Use that in the aforementioned two files

Reply:
Do you mean to get instance like this ?
 `AppMetaInfo appMetaInfo = AppMetaInfo.AppMetaInfo();`